### PR TITLE
Add OpenMP directives for ellblock

### PR DESCRIPTION
**Pull request by jeanlucf22**
Note: Done for most important functions only

---
This change is [Reviewable](https://reviewable.io/reviews/lanl/bml/397)

### CODE REVIEW

File: src/C-interface/ellblock/bml_multiply_ellblock_typed.c [hunk 1]
```diff
@@ -129,14 +129,30 @@ void *TYPED_FUNC(
     for (int ib = 0; ib < NB; ib++)
         maxbsize = MAX(maxbsize, bsize[ib]);
     int maxbsize2 = maxbsize * maxbsize;
-    REAL_T *x_ptr_storage = calloc(maxbsize2 * NB, sizeof(REAL_T));
-    for (int ib = 0; ib < NB; ib++)
-        x_ptr[ib] = &x_ptr_storage[ib * maxbsize2];
+    REAL_T *x_ptr_storage =
+        calloc(maxbsize2 * NB * omp_get_max_threads(), sizeof(REAL_T));
```
Comment (nicolasbock) [inline src/C-interface/ellblock/bml_multiply_ellblock_typed.c]: `omp_get_max_threads()` is defined in `omp.h` which is only available with OpenMP. You need to `ifdef` this somehow.

Comment (jeanlucf22) [reply src/C-interface/ellblock/bml_multiply_ellblock_typed.c]: Ooops, I missed that one. I thought our tests would catch that?

Comment (nicolasbock) [reply src/C-interface/ellblock/bml_multiply_ellblock_typed.c]: I am surprised that the serial test passed though: https://app.circleci.com/pipelines/github/lanl/bml/270/workflows/2ab58551-3414-4b9d-994a-44cf12484fa7/jobs/2588/steps

Comment (nicolasbock) [reply src/C-interface/ellblock/bml_multiply_ellblock_typed.c]: Yes, I was thinking the same. I'll check

Comment (jeanlucf22) [reply src/C-interface/ellblock/bml_multiply_ellblock_typed.c]: @nicolasbock You may be looking at an outdated file. The last commit should be fine

Comment (nicolasbock) [reply src/C-interface/ellblock/bml_multiply_ellblock_typed.c]: Ah yes, that was it :smile:

**Pull request merged**